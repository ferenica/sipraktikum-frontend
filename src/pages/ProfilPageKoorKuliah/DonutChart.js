import React, {Component} from 'react';
import ReactApexChart from 'react-apexcharts';
import Grid from '@material-ui/core/Grid';
import {Row, Col} from 'react-bootstrap';
import {MdTrendingUp} from 'react-icons/md';


/**
 *
 */
class DonutChart extends Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);

    this.state = {
      series: [],
      series2: [],
      options: {
        chart: {
          type: 'donut',
          width: '100%',
          height: '50%',
        },
        plotOptions: {
          pie: {
            startAngle: 0,
            endAngle: 360,
          },
        },
        dataLabels: {
          enabled: false,
        },
        legend: {
          show: false,
        },
        fill: {
          type: 'gradient',
          colors: ['#FF9C51', '#F2F2F2'],
        },
        responsive: [{
          options: {
            chart: {
              width: 1000,
            },
            legend: {
              show: false,
            },
          },
        }],
      },
    };
  }

  /**
   *
   */
  async getPostAPI() {
    try {
      const requestOptions = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
      };
      const awaitResponse = await fetch('http://ppl-berkah-backend.herokuapp.com/api/v1/koordinator-kuliah/persentase-pengumpulan-laporan', requestOptions)
          .then((response) => response.json())
          .then((data) => {
            // const arr = [];
            // Object.keys(data.data).forEach((key) => arr.push(data.data[key]));
            // const initialItems = arr;
            // const items = initialItems;
            // this.state.series = items;
            this.setState({
              MetaProducts: this.state,
              series: [data.data.this_week, 100-data.data.this_week],
              persen: data.data.this_week,
              diff: data.data.difference_percent,
              absolute: Math.abs(data.data.difference_percent),
              init: 1,
            });
          });
      return await awaitResponse;
    } catch (error) {
      alert(error);
    }
  }
  /**
   *
   */
  async getPostAPIPenilaian() {
    try {
      const requestOptions = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
      };
      const awaitResponse = await fetch('http://ppl-berkah-backend.herokuapp.com/api/v1/koordinator-kuliah/persentase-penilaian-laporan-supervisor-sekolah', requestOptions)
          .then((response) => response.json())
          .then((data) => {
            // const arr = [];
            // Object.keys(data.data).forEach((key) => arr.push(data.data[key]));
            // const initialItems = arr;
            // const items = initialItems;
            // this.state.series2 = items;
            this.setState({
              MetaProducts: this.state,
              series2: [data.data.this_week, 100-data.data.this_week],
              persenPenilaian: data.data.this_week,
              diffPenilaian: data.data.difference_percent,
              absolutePenilaian: Math.abs(data.data.difference_percent),
              init: 1,
            });
          });
      return await awaitResponse;
    } catch (error) {
      alert(error);
    }
  }

  /**
   *
   */
  componentDidMount() {
    this.getPostAPI();
    this.getPostAPIPenilaian();
  }

  /**
   *
   */
  render() {
    return (
      <>
        <Grid container xs={6} style={{border: '1px solid #DEDEDE', borderRadius: '5px'}}>
          <Col style={{alignItems: 'center', display: 'flex', marginLeft: '-5px'}}>
            <ReactApexChart options={this.state.options} series={this.state.series} type="donut" width='100%' height='85%' />
          </Col>
          <Col xs={8}>
            <Row style={{fontFamily: 'Nunito Sans', fontWeight: 'normal', fontSize: '16px', lineHeight: '22px', letterSpacing: '0.0025em'}}>
              {this.state.persen}% Mahasiswa mengumpulkan tepat waktu
            </Row>
            <Row>
              <Col xs={1}><MdTrendingUp color="#219653" size={15}></MdTrendingUp></Col>
              <Col xs={9} style={{fontFamily: 'Nunito Sans', fontWeight: 'normal', fontSize: '14px', color: '#AAAAAA', lineHeight: '22px'}}>
                {this.state.diff >= 0 ?
                <div>Meningkat {this.state.diff}% dari minggu sebelumnya</div> :
                <div>Menurun {this.state.absolute}% dari minggu sebelumnya</div>}
              </Col>
            </Row>
          </Col>
        </Grid>
        <Grid container xs={6} style={{border: '1px solid #DEDEDE', borderRadius: '5px'}}>
          <Col style={{alignItems: 'center', display: 'flex', marginLeft: '-5px'}}>
            <ReactApexChart options={this.state.options} series={this.state.series2} type="donut" width='100%' height='85%' />
          </Col>
          <Col xs={8}>
            <Row style={{fontFamily: 'Nunito Sans', fontWeight: 'normal', fontSize: '16px', lineHeight: '22px', letterSpacing: '0.0025em'}}>
              {this.state.persenPenilaian}% Laporan telah dinilai oleh Supervisor Sekolah
            </Row>
            <Row>
              <Col xs={1}><MdTrendingUp color="#219653" size={15}></MdTrendingUp></Col>
              <Col xs={9} style={{fontFamily: 'Nunito Sans', fontWeight: 'normal', fontSize: '14px', color: '#AAAAAA', lineHeight: '22px'}}>
                {this.state.diff >= 0 ?
                <div>Meningkat {this.state.diffPenilaian}% dari minggu sebelumnya</div> :
                <div>Menurun {this.state.absolutePenilaian}% dari minggu sebelumnya</div>}
              </Col>
            </Row>
          </Col>
        </Grid>
      </>
    );
  }
}

export default DonutChart;
