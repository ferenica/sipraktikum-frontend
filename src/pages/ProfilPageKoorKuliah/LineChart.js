import React, {Component} from 'react';
import ReactApexChart from 'react-apexcharts';

/**
 *
 */
class LineChart extends Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);

    this.state = {
      hasError: false,
      series: [{
        data: [1, 6, 10, 5, 6, 7],
      }],
      options: {
        chart: {
          height: 350,
          type: 'area',
        },
        dataLabels: {
          enabled: false,
        },
        stroke: {
          curve: 'smooth',
          colors: ['#FF8326', '#FF9C51'],
        },
        fill: {
          colors: ['#FF8326', '#FF9C51'],
        },
        markers: {
          colors: '#FF8326',
        },
        xaxis: {
          categories: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
          title: {
            text: 'Minggu ke',
            floating: false,
            offsetY: 13,
            style: {
              fontSize: '14px',
              fontWeight: 'normal',
              fontFamily: 'Nunito Sans',
              color: '#AAAAAA',
            },

          },
        },
        yaxis: {
          categories: [10, 20, 30, 40, 50, 60, 70, 80, 90],
          max: 90,
          floating: false,
          forceNiceScale: true,
          title: {
            text: 'Jumlah Mahasiswa',
            style: {
              fontSize: '14px',
              fontWeight: 'normal',
              fontFamily: 'Nunito Sans',
              color: '#AAAAAA',
            },
          },
        },
      },


    };
  }

  /**
   *
   */
  async getPostAPI() {
    try {
      const requestOptions = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
      };
      const awaitResponse = await fetch('http://ppl-berkah-backend.herokuapp.com/api/v1/koordinator-kuliah/riwayat-ketepatan-pengumpulan-laporan', requestOptions)
          .then((response) => response.json())
          .then((data) => {
            const arr = [];
            Object.keys(data.data).forEach((key) => arr.push(data.data[key]));
            const initialItems = arr;
            const items = initialItems;
            this.state.series = [{name: 'series1', data: items}];
            this.setState({
              MetaProducts: this.state,
              init: 1,
            });
          });
      return await awaitResponse;
    } catch (error) {
      alert(error);
    }
  }

  /**
   *
   */
  componentDidMount() {
    this.getPostAPI();
  }

  /**
   *
   * @param {*} error
   * @param {*} info
   */
  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({hasError: true});
  }

  /**
   *
   */
  renderChart() {
    return (
      <>
        <div id="chart">
          <ReactApexChart options={this.state.MetaProducts.options} series={this.state.MetaProducts.series} type="area" height={350} />
        </div>
      </>
    );
  }

  /**
   *
   */
  render() {
    return this.state.init ? <div>{this.renderChart()}</div> : <h1>Something went wrong.</h1>;
  }
}

export default LineChart;
