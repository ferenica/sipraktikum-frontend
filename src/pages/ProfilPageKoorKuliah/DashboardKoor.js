import React from 'react';
import LineChart from './LineChart';
import DonutChart from './DonutChart';
import Grid from '@material-ui/core/Grid';

/**
 * return for function dashboard
 * @return  {div} div to return
 */
export default function DashboardKoor() {
  return (
    <div>
      <br></br>
      <Grid xs={12} style={{display: 'flex', flexDirection: 'row'}}>
        <DonutChart></DonutChart>
      </Grid>
      <br></br>
      <h3 style={{textAlign: 'center'}}>
        <br></br>
      Grafik Ketepatan Waktu Pengumpulan Laporan
      </h3>
      {/* Disabled due to performance reason */}
      {/* <LineChart></LineChart> */}
    </div>
  );
}
