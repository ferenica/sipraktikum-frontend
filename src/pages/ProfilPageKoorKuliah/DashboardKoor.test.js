import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import {screen} from '@testing-library/dom';
import DashboardKoor from './DashboardKoor';
import LineChart from './LineChart';
import DonutChart from './DonutChart';

jest.mock('axios');

Enzyme.configure({adapter: new EnzymeAdapter()});


describe('dashboard koor praktikum page', () => {
  it('exist dashboard koor', () => {
    const wrapper = shallow(<DashboardKoor/>);
    expect(wrapper).toMatchSnapshot();
  });

  // it('text appear', () => {
  //   const wrap = shallow(<DashboardKoor/> );
  //   expect(wrap.text()).toEqual('<DonutChart />Grafik Ketepatan Waktu Pengumpulan Laporan<LineChart />');
  // });
  it('donut chart appear', () => {
    const wrap = shallow(<DashboardKoor/> );
    expect(
        wrap.containsMatchingElement(
            <DonutChart></DonutChart>,
        ),
    ).toBeTruthy();
  });
  // it('line chart appear', () => {
  //   const wrap = shallow(<DashboardKoor/> );
  //   expect(
  //       wrap.containsMatchingElement(
  //           <LineChart></LineChart>,
  //       ),
  //   ).toBeTruthy();
  // });
  it('no button exist', () => {
    const submitButton = screen.queryByText('submit');
    expect(submitButton).toBeNull();
  });
});

