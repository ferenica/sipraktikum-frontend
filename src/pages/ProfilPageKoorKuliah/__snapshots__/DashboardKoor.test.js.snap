// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard koor kuliah page exist daftar mahasiswa 1`] = `ShallowWrapper {}`;

exports[`dashboard koor kuliah page exist dashboard koor 1`] = `ShallowWrapper {}`;

exports[`dashboard koor praktikum page exist dashboard koor 1`] = `ShallowWrapper {}`;
